e2e/cypress/support/okta_commands.js

```diff
@@ -0,0 +1,233 @@
+    const profile = {
+        firstName: user.firstname,
+        lastName: user.lastname,
+        email: user.email,
+        login: user.email,
+        userType: user.userType,
+        isAdmin: user.isAdmin,
+        isGuest: user.isGuest
+    };
+    return profile;
+}
+
+Cypress.Commands.add('oktaCreateUser', (user = {}) => {
+    const profile = buildProfile(user);
+    cy.task('oktaRequest', {
+        baseUrl: Cypress.env('oktaApiUrl'),
+        urlSuffix: '/users/',
+        method: 'post',
+        token,
+        data: {
+            profile,
+            credentials: {
+                password: {value: user.password},
+                recovery_question: {
+                    question: 'What is the best open source messaging platform for developers?',
+                    answer: 'Mattermost'
+                }
+            }
+        }
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        const userId = response.data.id;
+        return userId;
+    });
+});
+
+Cypress.Commands.add('oktaGetUser', (userId = '') => {
+    cy.task('oktaRequest', {
+        baseUrl: Cypress.env('oktaApiUrl'),
+        urlSuffix: '/users?q=' + userId,
+        method: 'get',
+        token,
+    }).then((response) => {
+        //expect(response.status).to.be.equal(200);
+        if (response.data.length > 0) {
+            return response.data[0].id;
+        }
+        return null;
+    });
+});
+
+Cypress.Commands.add('oktaUpdateUser', (userId = '', user = {}) => {
+    const profile = buildProfile(user);
+
+    cy.task('oktaRequest', {
+        baseUrl: Cypress.env('oktaApiUrl'),
+        urlSuffix: '/users/' + userId,
+        method: 'post',
+        token,
+        data: {
+            profile
+        }
+    }).then((response) => {
+        expect(response.status).to.equal(201);
+        return cy.wrap(response.data);
+    });
+});
+
+//first we deactivate the user, then we actually delete it
+Cypress.Commands.add('oktaDeleteUser', (userId = '') => {
+    cy.task('oktaRequest', {
+        baseUrl: Cypress.env('oktaApiUrl'),
+        urlSuffix: '/users/' + userId,
+        method: 'delete',
+        token
+    }).then((response) => {
+        expect(response.status).to.equal(204);
+        expect(response.data).is.empty;
+        cy.task('oktaRequest', {
+            baseUrl: Cypress.env('oktaApiUrl'),
+            urlSuffix: '/users/' + userId,
+            method: 'delete',
+            token
+        }).then((_response) => {
+            expect(_response.status).to.equal(204);
+            expect(_response.data).is.empty;
+        });
+    });
+});
+
+Cypress.Commands.add('oktaDeleteSession', (userId = '') => {
+    cy.task('oktaRequest', {
+        baseUrl: Cypress.env('oktaApiUrl'),
+        urlSuffix: '/users/' + userId + '/sessions',
+        method: 'delete',
+        token
+    }).then((response) => {
+        expect(response.status).to.equal(204);
+        expect(response.data).is.empty;
```

Comment:
Can we also add a step to delete the session cookie on the Cypress browser. At times, if any of the test fails due to some issue, the user's previous session does not gets cleared and as a result, instead of displaying the Okta login screen, the user is directly redirected to the Mattermost page after logging in. 

```
    // Ensure we clear out these specific cookies
    ['JSESSIONID'].forEach((cookie) => {
        cy.clearCookie(cookie);
    }); 
```

Reply:
That should be covered by the MM logout actions (either from selecting the logout buttons/links action from the UX) or the cy.apiLogout() called as the first operation in the doSamlLogin() - it cleans the cookies. If would say let's hold on this unless we hit flakiness later on.

Reply:
Mattermost would only clean out the cookies which were created by the Mattermost server. But the cookie which I am referring to `JSESSIONID` is actually created by the Okta application. This cookie does not gets cleared at times when there is a failure, and hence instead of displaying the Okta login screen, the user is directly taken to the Mattermost logged in page.

Reply:
Thanks, wasn't aware of that - will make the change then.